proofreader/runner.py

```diff
@@ -0,0 +1,61 @@
+"""Module `runner`
+
+This module
+
+- checks for local config overrides
+- runs `flake8` and `pylint` on the
+- outputs run_command progress to stdout and stderr
+- returns exit code based on the success or failure of both commands
+
+"""
+from __future__ import print_function
+from subprocess import Popen
+import sys
+
+from proofreader.config.utils import (
+    get_flake8_options, get_pylint_options
+)
+
+
+def run(targets, config_dir='.'):
+    # type: (List[str], str) -> None
+    """Runs `pylint` and `flake8` commands and exits based off the evaluation
+    of both command results.
+
+    :param targets: List[str]
+    :param config_dir: str
+    :return:
+    """
+    pylint_options = get_pylint_options(config_dir=config_dir)
+    flake8_options = get_flake8_options(config_dir=config_dir)
+
+    pylint_return_state = _run_command(command='pylint', targets=targets, options=pylint_options)
+    flake8_return_state = _run_command(command='flake8', targets=targets, options=flake8_options)
+
+    if not flake8_return_state and not pylint_return_state:
+        sys.exit(0)
+    else:
```

Comment:
different exit codes would make the result unambiguous:
- `1`: called with bad arguments
- `2`: flake8 failing
- `3`: pylint failing

Reply:
Good idea. Will add it in.

Reply:
On review of the function, I already account for most of this. You can't call the `run()` function without the required `targets` value, it would fail before hand. 

If you do provide values but they are not valid they will still get passed to `flake8` and `pylint` which will cause them to fail and output why they failed e.g. `Directory does not exist` along with details. This would then cause a `sys.exit(1)` indicating a failure and displaying the output to the console as normal.

So the `run()` either succeeds or fails and if it fails it gives you the details why anyway, what failed and why.